File: MAP/LAB7-MAP/LAB7-MAP/Statements/PrintStmt.cs
using System;
using MAP_Lab7.Exceptions;
using MAP_Lab7.Expressions;

namespace MAP_Lab7.Statements
{
    public class PrintStmt:Statement
    {
        private IExpression expr;
        public PrintStmt(IExpression e)
        {
            expr = e;
        }

        public PrgState Execute(PrgState p)
        {
            try
            {
                int r = expr.Eval(p.Dict);
                p.OutputList.AddList(r);
            }
            catch (EvaluationException e)
            {
                Console.WriteLine(e.Message);
            }
            catch (InexistentSymbolException e2)
            {
                Console.WriteLine(e2.Message);
            }
            catch (DivByZeroException e3)
            {
                Console.WriteLine(e3.Message);
            }
            return p;
        }

        public override string ToString()
        {
            return "" + expr;
        }
    }
}

File: MAP/Lab3/src/Model/Statement/PrintStmt.java
package Model.Statement;

import Model.Expression.Expression;
import Exception.DivByZeroException;
import Exception.EvaluationException;
import Exception.InexistentSymbolException;

public class PrintStmt implements Statement
{
    private Expression expr;
    public PrintStmt(Expression e)
    {
        expr = e;
    }

    @Override
    public PrgState execute(PrgState p)
    {
        try
        {
            int r = expr.Eval(p.getSymbolT());
            p.getList().add(r);
        }
        catch(EvaluationException | DivByZeroException | InexistentSymbolException e)
        {
            System.out.println(e.getMessage());
        }
        return p;
    }

    @Override
    public String toString() {
        return " " + "Print( " + expr + " )";
    }
}
